superset/db_engine_specs/pinot.py

```diff
@@ -86,7 +86,9 @@ def get_timestamp_expr(
                 python_pattern,
                 java_pattern,
             ) in cls._python_to_java_time_patterns.items():
-                java_date_format.replace(python_pattern, java_pattern)
+                java_date_format = java_date_format.replace(
+                    python_pattern, java_pattern
+                )
```

Comment:
Line 84 won't work correctly for `pdf = None` (I assume `pdf` will always be defined, but it doesn't look right). Let's fix that now, too, something like
```python
            if pdf:
                java_date_format = pdf
                for (
                    python_pattern,
                    java_pattern,
                ) in cls._python_to_java_time_patterns.items():
                    java_date_format = java_date_format.replace(
                        python_pattern, java_pattern
                    )
```

Reply:
Thanks! Fixed.